### ISSUE TITLE
Kubernetes Scanning with Gitlab CI

### ISSUE THREAD

**Issue by sherifkayad** [aquasecurity/trivy#3234]
Hello,

Currently using the command below:
```bash
trivy k8s -n MY_NAMESPACE all --report all --format template --template '@gitlab-ci.tpl'
```

Yields the following error:
```
FATAL   unable to write results: unknown format "template". Use "json" or "table"
```

Is it possible to maybe add a Gitlab CI Template or the instructions how the scan of a whole cluster and posting a report to Gitlab-CI could be made possible?

**Comment by empeekdev** [comment]
https://github.com/aquasecurity/trivy/issues/2661

Format template is not supported yet for Kubernetes scanning

**Comment by sherifkayad** [comment]
Keeping this one alive

**Comment by sherifkayad** [comment]
Keep alive

**Comment by itaysk** [comment]
@chen-keinan is it possible to refactor the code such that the output flag is just using the trivy one? there will be more improvements to output formatting soon and we want all the scanners in trivy to benefit from it

**Comment by knqyf263** [comment]
I don't think it is easy to use the trivy output implementation because the result of Kubernetes scanning has a different structure.

**Comment by jkroepke** [comment]
Would it be possible to enable the template output, if `--scanners vuln` is configured?
